test/EndToEndTests/Tests/Client/Microsoft.OData.Client.E2E.Tests/ActionOverloadingTests/Tests/ActionOverloadingQueryTests.cs

```diff
@@ -0,0 +1,205 @@
+using Microsoft.OData.Client.E2E.Tests.Common.Clients.EndToEnd.Default;
+using Microsoft.OData.Client.E2E.Tests.Common.Server.EndToEnd;
+using Microsoft.OData.Edm;
+using Xunit;
+
+namespace Microsoft.OData.Client.E2E.Tests.ActionOverloadingTests.Tests
+{
+    public class ActionOverloadingQueryTests : EndToEndTestBase<ActionOverloadingQueryTests.TestsStartup>
+    {
+        private readonly Uri _baseUri;
+        private readonly Container _context;
+        private IEdmModel _model = null;
+        public const string PersonTypeName = "Microsoft.OData.Client.E2E.Tests.Common.Server.EndToEnd.Person";
+        public const string EmployeeTypeName = "Microsoft.OData.Client.E2E.Tests.Common.Server.EndToEnd.Employee";
+        public const string SpecialEmployeeTypeName = "Microsoft.OData.Client.E2E.Tests.Common.Server.EndToEnd.SpecialEmployee";
+        public const string ContractorTypeName = "Microsoft.OData.Client.E2E.Tests.Common.Server.EndToEnd.Contractor";
+        public const string ProductTypeName = "Microsoft.OData.Client.E2E.Tests.Common.Server.EndToEnd.Product";
+        public const string OrderLineTypeName = "Microsoft.OData.Client.E2E.Tests.Common.Server.EndToEnd.OrderLine";
+        public const string MetadataPrefix = "$metadata#Default.";
+        public const string ContainerPrefix = "#Default.";
+
+        public class TestsStartup : TestStartupBase
+        {
+            public override void ConfigureServices(IServiceCollection services)
+            {
+                services.ConfigureControllers(typeof(ProductsController), typeof(OrderLinesController), typeof(PeopleController), typeof(MetadataController));
+
+                services.AddControllers().AddOData(opt => opt.Count().Filter().Expand().Select().OrderBy().SetMaxTop(null)
+                    .AddRouteComponents("odata", CommonEndToEndEdmModel.GetEdmModel()));
+            }
+        }
+
+        public ActionOverloadingQueryTests(TestWebApplicationFactory<TestsStartup> fixture)
+            : base(fixture)
+        {
+            _baseUri = new Uri(Client.BaseAddress, "odata/");
+            _context = new Container(_baseUri);
+            _context.HttpClientFactory = HttpClientFactory;
+            _model = CommonEndToEndEdmModel.GetEdmModel();
+        }
+
+        /// <summary>
+        /// Verifies that the specified operations are present in the test service model.
+        /// </summary>
+        [Fact]
+        public void VerifyServiceOperationsInModel()
+        {
+            var expectedOperations = new List<string>
+            {
+                "UpdatePersonInfo",
+                "RetrieveProduct"
+            };
+
+            var actionImports = _model.EntityContainer.OperationImports();
+
+            foreach (var operationName in expectedOperations)
+            {
+                var filteredImports = actionImports.Where(fi => fi.Name == operationName);
+                Assert.NotNull(filteredImports);

```

Comment:
Even if `filteredImports` was empty, this assertion would not fail, because the return of `Where()` clause would be a non-null object, even if it represent an empty collection.

Maybe should use `Assert.True(actionImports.Any(fi => fi.Name == operationName)` if you want to use not null:

```c#
var import = actionImports.FirstOrDefault(fi => fi.Name == operationName);
Assert.NotNull(import);
```